peps/pep-0733.rst

```diff
@@ -0,0 +1,657 @@
+---------------------------------------
+
+Applications with an embedded Python interpreter. Examples are
+`Blender <https://docs.blender.org/api/current/info_overview.html>`__ and
+`OBS <https://obsproject.com/wiki/Getting-Started-With-OBS-Scripting>`__.
+
+They need to be able to:
+
+* Configure the interpreter (import paths, inittab, ``sys.argv``, etc.).
+* Interact with the execution model and program lifetime, including
+  clean interpreter shutdown and restart.
+* Represent complex data models in a way Python can use without
+  having to create deep copies.
+* Provide and import frozen modules.
+* Run multiple independent interpreters (in particular, when embedded
+  in a library that wants to avoid global effects).
+
+Python Implementations
+----------------------
+
+Python implementations such as
+`CPython <https://www.python.org>`__,
+`PyPy <https://www.pypy.org>`__,
+`GraalPy <https://www.graalvm.org/python/>`__,
+`IronPython <https://ironpython.net>`__,
+`RustPython <https://github.com/RustPython/RustPython>`__,
+`MicroPython <https://micropython.org>`__,
+and `Jython <https://www.jython.org>`__), may take
+very different approaches for the implementation of
+different subsystems. They need:
+
+* The API to be abstract and hide implementation details.
+* A specification of the API, ideally with a test suite
+  that ensures compatibility.
+* It would be nice to have an ABI that can be shared
+  across Python implementations.
+
+Alternative APIs
+----------------
+
+There are several projects that implement alternatives to the
+C API, which offer extension users advantanges over programming
+directly with the C API. These APIs are implemented with the
+C API, and in some cases by using CPython internals.
+Examples are
+`HPy <https://hpyproject.org>`__ and
+`pythoncapi-compat <https://pythoncapi-compat.readthedocs.io/en/latest/>`__.
+CPython's DSL for parsing function arguments, the
+`Argument Clinic <https://devguide.python.org/development-tools/clinic/>`__,
+can also be seen as belonging to this category of stakeholders.
+
+Such systems need minimal building blocks for accessing CPython
+efficiently. They don't necessarily need an ergonomic API, because
+they typically generate code that is not intended to be read
+by humans. But they do need it to be comprehensive enough so that
+they don't need to access internals, without sacrificing performance.
+
+These tools might also benefit from a choice between a more stable
+and a more performant (possibly lower-level) API. Their users could
+then decide whether they can afford to regenerate the code often or
+trade some performance for more stability and less maintenance work.
+
+
+Binding Generators
+------------------
+
+Libraries that create bindings between Python and other object models,
+paradigms or languages, such as
+`Cython <https://cython.org>`__,
+`cffi <https://cffi.readthedocs.io/>`__,
+`pybind11 <https://pybind11.readthedocs.io/en/stable/>`__ for C++11,
```

Comment:
```suggestion
`pybind11 <https://pybind11.readthedocs.io/en/stable/>`__ for C++11,
`nanobind <https://github.com/wjakob/nanobind>`__ for C++11,
```

Reply:
Perhaps combine the two into a single line: "pybind11 and nanobind for C++11"? The links will render as a single word, and this list is only examples, not exhaustive.

Reply:
nanobind requires C++17, so no.

Reply:
Perhaps just "for C++" then? Again, examples, not a table of all the bindings that exist.